Gems/Atom/RPI/Code/Source/RPI.Public/Pass/AttachmentReadback.cpp

```diff
@@ -432,37 +386,107 @@ namespace AZ
+                        if (m_state == ReadbackState::Reading)
                         {
-                            m_state = ReadbackState::Success;
+                            if (CopyBufferData(readbackBufferCurrentIndex, deviceIndex))
+                            {
+                                m_state = ReadbackState::Success;
+                            }
+                            else
+                            {
+                                m_state = ReadbackState::Failed;
+                            }
                         }
-                        else
+                        if (m_callback)
                         {
-                            m_state = ReadbackState::Failed;
+                            m_callback(GetReadbackResult());
                         }
-                    }
-                    if (m_callback)
+
+                        Reset();
+                    });
+
+            for (uint16_t itemIdx = 0; itemIdx < static_cast<uint16_t>(m_readbackItems.size()); itemIdx++)
+            {
+                auto& readbackItem = m_readbackItems[itemIdx];
+                if (m_attachmentType == RHI::AttachmentType::Image)
+                {
+                    // copy image to read back buffer since only buffer can be accessed by host
+                    const auto image = readbackItem.m_copyItem.m_mdImage.m_mdSourceImage;
+                    if (!image)
                     {
-                        m_callback(GetReadbackResult());
+                        AZ_Warning(
+                            "AttachmentReadback",
+                            false,
+                            "Failed to find attachment image %s for copy to buffer",
+                            m_copyAttachmentId.GetCStr());
+                        return;
                     }
 
-                    Reset();
+                    m_imageDescriptor = image->GetDescriptor();
+                    // [GFX TODO] [ATOM-14140] [Pass Tree] Add the ability to output all the array subresources and planars
+                    // only array 0, and one aspect (planar) at this moment.
+                    // Note: Mip Levels and Texture3D images are supported.
+                    const uint16_t mipSlice = m_imageMipsRange.m_mipSliceMin + itemIdx;
+                    RHI::ImageSubresourceRange range(mipSlice, mipSlice, 0, 0);
+                    range.m_aspectFlags = RHI::ImageAspectFlags::Color;
+
+                    // setup aspect
+                    RHI::ImageAspect imageAspect = RHI::ImageAspect::Color;
+                    RHI::ImageAspectFlags imageAspectFlags = RHI::GetImageAspectFlags(m_imageDescriptor.m_format);
+                    if (RHI::CheckBitsAll(imageAspectFlags, RHI::ImageAspectFlags::Depth))
+                    {
+                    imageAspect = RHI::ImageAspect::Depth;
+                    range.m_aspectFlags = RHI::ImageAspectFlags::Depth;
```

Comment:
something is wonky here with the indentation

Reply:
Fixed